src/types/tuple.rs

```diff
@@ -391,6 +437,51 @@ impl DoubleEndedIterator for BoundTupleIterator<'_> {
             None
         }
     }
+
+    #[inline]
+    #[cfg(not(feature = "nightly"))]
+    fn nth_back(&mut self, n: usize) -> Option<Self::Item> {
+        let length_size = self.length.min(self.tuple.len());
+        if self.index + n < length_size {
+            let target_index = length_size - n - 1;
+            let item = {
+                {
+                    unsafe {
+                        BorrowedTupleIterator::get_item(self.tuple.as_borrowed(), target_index)
+                            .to_owned()
+                    }
+                }
+            };
+            self.length = target_index;
+            Some(item)
+        } else {
+            None
+        }
+    }
+
+    #[inline]
+    #[cfg(feature = "nightly")]
+    fn advance_back_by(&mut self, n: usize) -> Result<(), NonZero<usize>> {
+        let max_len = self.length.min(self.tuple.len());
+        let currently_at = self.index;
+        if currently_at >= max_len {
+            if n == 0 {
+                return Ok(());
+            } else {
+                return Err(unsafe { NonZero::new_unchecked(n) });
+            }
+        }
+
+        let items_left = max_len - currently_at;
+        if n <= items_left {
+            self.length = max_len - n;
+            Ok(())
+        } else {
+            self.length = max_len;
```

Comment:
```suggestion
            self.length = currently_at;
```
I believe this is wrong. The iterator should be empty after this case, right? Let's add this to the tests below as well.

Reply:
Indeed since it's backward iteration. I've adapted your suggestions (and also reflect the same for `BoundListIterator`. Thanks a lot!